File: iosApp/iosApp/feature/address/create_address/CreateAddressViewState.swift
//
//  CreateAddressViewState.swift
//  iosApp
//
//  Created by Марк Шавловский on 07.02.2024.
//  Copyright © 2024 orgName. All rights reserved.
//

import Foundation
import SwiftUI

struct CreateAddressViewState {
    var street: String
    var streetError: LocalizedStringKey?
    var streetSuggestionList: [StreetItem]
    var isSuggestionLoading: Bool
    var house: String
    var houseError: LocalizedStringKey?
    var flat: String
    var entrance: String
    var floor: String
    var comment: String
    var isCreateLoading: Bool
}

File: app/src/main/java/com/bunbeauty/papakarlo/feature/address/screen/createaddress/CreateAddressViewState.kt
package com.bunbeauty.papakarlo.feature.address.screen.createaddress

import androidx.annotation.StringRes
import androidx.compose.runtime.Immutable
import com.bunbeauty.shared.presentation.SuggestionUi
import com.bunbeauty.shared.presentation.base.BaseViewState
import kotlinx.collections.immutable.ImmutableList

@Immutable
data class CreateAddressViewState(
    val street: String,
    @StringRes val streetErrorStringId: Int?,
    val streetSuggestionList: ImmutableList<SuggestionUi>,
    val isSuggestionLoading: Boolean,
    val house: String,
    @StringRes val houseErrorStringId: Int?,
    val flat: String,
    val entrance: String,
    val floor: String,
    val comment: String,
    val isCreateLoading: Boolean,
) : BaseViewState {
    val suggestionListNotEmpty = streetSuggestionList.isNotEmpty()
}
